pkg/service/http_service.go

```diff
@@ -19,153 +26,169 @@ type HTTPServiceConfiguration struct {
 
-	result, reason, err := s.eval.ResolveObjectValue(flagKey, params.DefaultValue.AdditionalProperties, contextObj)
+func (s HTTPService) ResolveNumber(
+	ctx context.Context,
+	req *gen.ResolveNumberRequest,
+) (*gen.ResolveNumberResponse, error) {
+	res := gen.ResolveNumberResponse{}
+	result, variant, reason, err := s.eval.ResolveNumberValue(req.GetFlagKey(), req.GetContext())
 	if err != nil {
-		handleEvaluationError(err, reason, w)
-		return
+		return &res, handleEvaluationError(err, reason)
 	}
-	_ = json.NewEncoder(w).Encode(gen.ResolutionDetailsObject{
-		Value: gen.ResolutionDetailsObject_Value{
-			AdditionalProperties: result,
-		},
-		Reason: &reason,
-	})
+	res.Reason = reason
+	res.Value = result
+	res.Variant = variant
+	return &res, nil
 }
 
-func (h *HTTPService) Serve(ctx context.Context, eval eval.IEvaluator) error {
-	if h.HTTPServiceConfiguration == nil {
-		return errors.New("http service configuration has not been initialised")
+func (s HTTPService) ResolveObject(
+	ctx context.Context,
+	req *gen.ResolveObjectRequest,
+) (*gen.ResolveObjectResponse, error) {
+	res := gen.ResolveObjectResponse{}
+	result, variant, reason, err := s.eval.ResolveObjectValue(req.GetFlagKey(), req.GetContext())
+	if err != nil {
+		return &res, handleEvaluationError(err, reason)
 	}
-	http.Handle("/", gen.Handler(Server{eval}))
-	_ = http.ListenAndServe(fmt.Sprintf(":%d", h.HTTPServiceConfiguration.Port), nil)
-
-	<-ctx.Done()
-	return nil
-}
-
-func handleParseError(err error, w http.ResponseWriter) {
-	log.Errorf("Error parsing context from body: %s", err.Error())
-	reason := model.ErrorReason
-	errorCode := model.ParseErrorCode
-	w.WriteHeader(400)
-	if err = json.NewEncoder(w).Encode(gen.ResolutionDetailsWithError{
-		ErrorCode: &errorCode,
-		Reason:    &reason,
-	}); err != nil {
-		log.Errorf("Error encoding response: %s", err.Error())
+	val, err := structpb.NewStruct(result)
+	if err != nil {
+		return &res, handleEvaluationError(err, reason)
 	}
+	res.Reason = reason
+	res.Value = val
+	res.Variant = variant
+	return &res, nil
 }
 
 // some basic mapping of errors from model to HTTP
-func handleEvaluationError(err error, reason string, w http.ResponseWriter) {
+func handleEvaluationError(err error, reason string) error {
 	// TODO: we should consider creating a custom error that includes a code instead of using the message for this.
+	statusCode := codes.Internal
 	message := err.Error()
 	switch message {
 	case model.FlagNotFoundErrorCode:
-		w.WriteHeader(404)
+		statusCode = codes.NotFound
 	case model.TypeMismatchErrorCode:
-		w.WriteHeader(400)
-	default:
-		w.WriteHeader(500)
+		statusCode = codes.InvalidArgument
 	}
 	log.Error(message)
-	if err = json.NewEncoder(w).Encode(gen.ResolutionDetailsWithError{
-		ErrorCode: &message,
-		Reason:    &reason,
-	}); err != nil {
-		log.Errorf("Error encoding response: %s", err.Error())
-	}
+	return status.Error(statusCode, message)
+}
+
+// TODO: could be replaced with a logging client
```

Comment:
Is there a issue for this? If not, please create one so we don't forget about it.

Reply:
https://github.com/open-feature/flagd/issues/29

I think it falls under this issue, and will add a link to this comment so that it is not missed